applications/console_text_messenger/src/main.rs

```diff
@@ -214,9 +218,33 @@ pub fn main() {
     let secret_key = CommsSecretKey::from_hex(settings.secret_key.unwrap().as_str()).unwrap();
     let public_key = CommsPublicKey::from_secret_key(&secret_key);
 
-    // TODO Use a less hacky crate to determine the local machines public IP address. This only works on Unix systems!
-    let ip = local_ip::get().expect("Could not determine local machines public IP address");
-    let local_net_address = match format!("{}:{}", ip, settings.control_port.unwrap()).parse() {
+    // get and filter interfaces
+    let interfaces: Vec<NetworkInterface> = datalink::interfaces()
+        .into_iter()
+        .filter(|interface| {
+            !interface.is_loopback() && interface.is_up() && interface.ips.iter().any(|addr| addr.is_ipv4())
+        })
+        .collect();
+
+    // select first interface
+    if interfaces.first().is_none() {
+        error!(
+            target: LOG_TARGET,
+            "No available network interface with an Ipv4 Address."
+        );
+        std::process::exit(1);
+    }
+
+    // get network interface and retrieve ipv4 address
+    let interface = interfaces.first().unwrap().clone();
+    let mut local_ip = String::new();
+    for addr in interface.ips {
```

Comment:
Don't you want to stop at the first suitable address here, rather than looping through them all?

`let Some(local_ip) = interface.ips.iter().find_map(|addr| addr.is_ipv4()) {...`

Reply:
Have updated it. Thanks 👍